src/curve/stable/StableMintBurn.sol

```diff
@@ -3,55 +3,68 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.0;
 
+import { stdMath } from "forge-std/Test.sol";
+import { SafeCast } from "@openzeppelin/utils/math/SafeCast.sol";
+
 import { Bytes32Lib } from "src/libraries/Bytes32.sol";
 import { FactoryStoreLib } from "src/libraries/FactoryStore.sol";
 import { StableMath } from "src/libraries/StableMath.sol";
 import { StableOracleMath } from "src/libraries/StableOracleMath.sol";
+import { LogCompression } from "src/libraries/LogCompression.sol";
 
-import { IReservoirCallee } from "src/interfaces/IReservoirCallee.sol";
-
+import { IAssetManager } from "src/interfaces/IAssetManager.sol";
+import { IAssetManagedPair } from "src/interfaces/IAssetManagedPair.sol";
 import { GenericFactory } from "src/GenericFactory.sol";
-import { ReservoirPair, Observation } from "src/ReservoirPair.sol";
-
-struct AmplificationData {
-    /// @dev initialA is stored with A_PRECISION (i.e. multiplied by 100)
-    uint64 initialA;
-    /// @dev futureA is stored with A_PRECISION (i.e. multiplied by 100)
-    uint64 futureA;
-    /// @dev initialATime is a unix timestamp and will only overflow every 584 billion years
-    uint64 initialATime;
-    /// @dev futureATime is a unix timestamp and will only overflow every 584 billion years
-    uint64 futureATime;
-}
+import { ReservoirERC20, ERC20 } from "src/ReservoirERC20.sol";
+import { Slot0, Observation } from "src/ReservoirPair.sol";
+import { StablePair, AmplificationData } from "src/curve/stable/StablePair.sol";
 
-contract StableMintBurn is ReservoirPair {
+contract StableMintBurn is ReservoirERC20, IAssetManagedPair {
```

Comment:
Do we need to implement the interface?

Reply:
we need to implement the interface cuz `IAssetManager` requires a `IAssetManagedPair` in the `getBalance` function. I considered the alternative, which is to change the interface, then we'll have to change a lot of existing code, which is really annoying